### PR TITLE
Testing Clients Added

### PR DESCRIPTION
**Pull request by dstarner**
## Description

Test clients are added for RPC and HTTP to more easily set up integration tests with Manifold.

**Change Log**

1. `RpcClient`, `HttpClient`, and `setup_test_env` added to aid in integration testing with callable clients.
1. Fixed a bug with the HTTP wrapping URLs that was caused by not decoding the request body.
1. Documentation added for the new features.

## Test Plan

Locally tested and linted

### Race Conditions / Edge Cases

N/A

### CODE REVIEW

File: tests/test_testing.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import types
+
```
Comment (Parushgit) [inline tests/test_testing.py]: I would prefer renaming this file with `testing.py` and actual `testing.py` with something like `testing_setup.py` if I got it right. It just makes unnecessary confusion.
